website/source/docs/guides/containers-guide.md

```diff
@@ -0,0 +1,328 @@
+
+## Prerequisites
+
+### Docker
+
+You will need a local install of Docker running on your machine for this guide. You can find the instructions for installing Docker on your specific operating system [here](https://docs.docker.com/install/).
+
+### Consul (Optional)
+
+If you would like to interact with your containerized Consul agents using a local install of Consul, follow the instructions [here](https://www.consul.io/docs/install/index.html) to install a local binary somewhere on your PATH.
+
+## Get the Docker Image
+
+First, pull the latest image. You will use Consul's official Docker image in this guide.
+
+```sh
+$ docker pull consul
+```
+
+Check the image was downloaded by listing all your docker images and looking for Consul image.
+
+```sh
+$ docker image ls
+REPOSITORY     TAG      IMAGE ID        CREATED             SIZE
+consul         latest   c836e84db154     4 days ago         107MB
+```
+## Configure and Run a Consul Server
+
+Next, you will use Docker command-line flags to start the agent as a server, configure networking, and bootstrap the cluster when one server is up.
+
+```sh
+$ docker run \
+    -d \
+    -p 8500:8500 \
+    -p 8600:8600/udp \
+    --name=badger \
+    consul agent -server -ui -node=server-1 -bootstrap-expect=1 -client=0.0.0.0
+```
+
+Since you started the container in detached mode, `-d`, the process will run in the background. You also set port mapping to your local machine as well as binding the cient interface of our agent to 0.0.0.0. This allows you to work directly with the Consul cluster from your local machine and to access Consul's UI and DNS over localhost. Finally, you are using Docker's default bridge network.
+
+Note, the Consul Docker image sets up the Consul configuration directory at `/consul/config` by default. The agent will load any configuration files placed in that directory. 
+
+~> The configuration directory is **not** exposed as a volume and will not persist data. Consul uses it only during startup and does not store any state there. 
+
+For more advanced use cases you can pass configuration JSON to `/consul/config` via the environment variable `CONSUL_LOCAL_CONFIG`, change the agent configuration with environment variables, or edit the configuration files directly. To learn more about configuring Consul see the [configuration documentation](https://www.consul.io/docs/agent/options.html). 
+
+### Discover the Server IP Address
+
+You can find the IP address of the Consul server by executing the `consul members` command inside of the `badger` container. 
+
+```sh
+$  docker exec badger consul members
+Node       Address         Status    Type    Build  Protocol  DC   Segment
+server-1  172.17.0.2:8301  alive     server  1.4.4  2         dc1  <all>
+```
+
+## Configure and Run a Consul Client
+
+Next, deploy a containerized Consul client and instruct it to join the server by giving it the server's IP address. Do not use detached mode, so you can see the startup output. 
```

Comment:
Small edit since the startup output isn't really what we're after.
```suggestion
Next, deploy a containerized Consul client and instruct it to join the server by giving it the server's IP address. Do not use detached mode, so you can reference the client logs during later steps. 
```